protocols/autonat/src/behaviour/as_client.rs

```diff
@@ -278,12 +278,9 @@ impl<'a> AsClient<'a> {
             log::debug!("Outbound dial-back request aborted: No dial-back addresses.");
             return Err(OutboundProbeError::NoAddresses);
         }
-        let server = match self.random_server() {
-            Some(s) => s,
-            None => {
-                log::debug!("Outbound dial-back request aborted: No qualified server.");
-                return Err(OutboundProbeError::NoServer);
-            }
+        let Some(server) = self.random_server() else {
```

Comment:
This could just be `unwrap_or()?`.

Reply:
We lose the `log` but we are returning an error that carries the same information so that is fine.